bluepysnap/circuit_validation.py

```diff
@@ -206,6 +207,40 @@ def _get_population_groups(population_h5):
             if isinstance(population_h5[name], h5py.Group) and name.isdigit()]
 
 
+def _sonata_group_to_dataframe(group, types_file, type_id, id_, group_id, group_index):
+    """Transforms hdf5 population group to pandas DataFrame.
+
+    Args:
+        group: HDF5 group of edges or nodes population
+        types_file: path to either 'node_types_file' or 'edge_types_file'
+        type_id: 'node_type_id' or 'edge_type_id' HDF5 dataset
+        id_: 'node_id' or 'edge_id' HDF5 dataset
+        group_id: 'node_group_id' or 'edge_group_id' HDF5 dataset
+        group_index: 'node_group_index' or 'edge_group_index' HDF5 dataset
+
+    Returns:
+        pd.DataFrame: dataframe with all group attributes
+    """
+    df = pd.DataFrame(type_id, columns=['type_id'])
+    size = df.size
+    df['node_id'] = np.arange(size) if id_ is None else id_
+    df['group_id'] = 0 if group_id is None else group_id
+    df['group_index'] = np.arange(size) if group_index is None else group_index
+    df = df[df['group_id'] == int(str(_get_group_name(group)))]
+    for k, v in group.items():
+        if isinstance(v, h5py.Dataset):
+            df[k] = v[:]
+    if '@library' in group:
+        for k, v in group['@library'].items():
+            if isinstance(v, h5py.Dataset):
+                df[k] = v[:][df[k].to_numpy(dtype=int)]
+    if types_file is None:
+        return df
+    types = pd.read_csv(types_file, sep=r'\s+')
+    types.rename(columns={types.columns[0]: 'type_id'}, inplace=True)
+    return pd.merge(df, types, on='type_id', how='left')
```

Comment:
that s cool it makes a first impl of the csv !

Reply:
Now we can validate Allen circuits as they heavily rely on this `.csv`.